aruco_detection/src/aruco_id_node.cpp

```diff
@@ -0,0 +1,83 @@
+#include "aruco_id_node.hpp"
+
+ArucoIdNode::ArucoIdNode() : loop_rate{10}
+{
+	dictionary = new cv::aruco::Dictionary;
+	// dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_5X5_100); //homemade docking plate
+	dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_ARUCO_ORIGINAL); // TAC
+
+	opImageSub = node.subscribe("filtered_image", 10, &ArucoIdNode::callback, this);
+	opImagePub = node.advertise<sensor_msgs::Image>("aruco_image", 100);
+	opIdPub    = node.advertise<std_msgs::Int64>("pipeline_id", 100);
+}
+
+void ArucoIdNode::callback(const sensor_msgs::ImageConstPtr &img_source)
+{
+	const cv_bridge::CvImageConstPtr cvImage = cv_bridge::toCvShare(img_source, sensor_msgs::image_encodings::BGR8);
+
+	cv::Mat img = cvImage->image;
+
+	std::vector<std::vector<cv::Point2f>> corners;
+	std::vector<int> ids;
+
+	// Detect markers
+	cv::aruco::detectMarkers(img, dictionary, corners, ids);
+	// Draw markers on image and publish image
+	cv::aruco::drawDetectedMarkers(img, corners, ids);
+	publishCVImg(img, cvImage->header.stamp);
+
+	if (ids.size() == 0)
+		return;
+
+	// Save found markers
+	// if (ids.size() > 1)
+	// ROS_WARN("More than one id detected!");
+	int foundId = ids.at(0);
+
+	// if (std::find(storedIds.begin(), storedIds.end(), foundId) != storedIds.end())
+	// 	return; // No new id detected
```

Comment:
rm commented out code

Reply:
Commented in instead. This was used during pipeline following to ensure unique ids were saved to file.

```suggestion
	// Ignore duplicate ids
	if (std::find(storedIds.begin(), storedIds.end(), foundId) != storedIds.end())
```

Reply:
was commented out for the visual inspection task as there could be duplicates there